app/viewmodels/CalculationPageHelper.scala

```diff
@@ -54,15 +56,22 @@ class CalculationPageHelper @Inject()(implicit val appConfig: AppConfig) extends
     dateToString(date)
   }
 
-  def isTTPActive(penaltyDetails: GetPenaltyDetails): Boolean = {
+  def isTTPActive(penaltyDetails: GetPenaltyDetails, vrn: String): Boolean = {
+    val currentDate = getFeatureDate
     penaltyDetails.latePaymentPenalty.exists {
       _.details.exists { //Current understanding is that TTP values are replicated across every LPP
         _.LPPDetailsMetadata.timeToPay.exists {
           _.exists(ttp => {
-            if (ttp.TTPEndDate.isDefined) {
-              (ttp.TTPStartDate.isEqual(getFeatureDate) || ttp.TTPStartDate.isBefore(getFeatureDate)) && (ttp.TTPEndDate.get.isEqual(getFeatureDate) || ttp.TTPEndDate.get.isAfter(getFeatureDate))
-            } else {
-              ttp.TTPStartDate.isEqual(getFeatureDate) || ttp.TTPStartDate.isBefore(getFeatureDate)
+            (ttp.TTPStartDate, ttp.TTPEndDate) match {
+              case (Some(startDate), Some(endDate)) =>
+                (startDate.isEqual(currentDate) || startDate.isBefore(currentDate)) && (endDate.isEqual(currentDate) || endDate.isAfter(currentDate))
+              case (None, Some(endDate)) => endDate.isEqual(currentDate) || endDate.isAfter(currentDate)
```

Comment:
Just reading the requirement on Jira ticket, it says after today, so is current date equality correct here?

Reply:
I would assume thats the case, feel free to raise it with BA / PM though. The previous logic did that but on the start date field (so I would assume dates are inclusive)